app/(auth)/login/forgot-password/page.tsx

```diff
@@ -0,0 +1,78 @@
+import Link from 'next/link';
+import { redirect } from 'next/navigation';
+import newServerClient from '@/supabase/utils/newServerClient';
+import { headers } from 'next/headers';
+
+export default async function ForgotPassword({
+  searchParams,
+}: {
+  searchParams: { message: string };
+}) {
+  const supabase = newServerClient();
+
+  // Access request headers to get the origin
+  let origin = headers().get('origin');
+  if (!origin) {
+    origin = 'http://localhost:3000/'; // Default to localhost if origin is not present
```

Comment:
can we just remove comments from here? As a convention we try to avoid comments unless we're adding documentation to the code but in this case that doesn't seem necessary